Automattic/wp-calypso
# Initial implementation of subtabs in the site management panel

Related to:

- https://github.com/Automattic/wp-calypso/issues/95385

## Proposed Changes

This is an initial implementation of hosting dashboard subtabs. The goal is to ship the subtab infrastructure first, so that we can build on top of it for each tab in parallel.

* Moved (renamed) `client/hosting/sites` to `client/sites`, which contains the site dashboard "infractructure".
  * My idea is that when we later move a tab to the new IA structure, we do it by:
    * moving it from `client/hosting/*` to `client/sites/*`, and
    * updating the URL (e.g. `/sites/marketing/:slug`)
    * Let me know if this refactor does no make sense or you have strong objection.
* Added capability for a panel to have sidebar containing subtabs.
* Implemented a dummy Marketing tab containing two subtabs. It's under the `untangling/hosting-menu` flag, which is set to true in dev and wpcalypso envs.

<img width="500" alt="image" src="https://github.com/user-attachments/assets/27e7f637-5176-4fdf-b61f-85c83a8c7f03">

## Why are these changes being made?

We want to migrate the pages from the Hosting menu in the left sidebar, to the "global" site-specific hosting dashboard panels.

## Testing Instructions

1. Use the provided Calypso link.
2. Go to /sites then click any site.
3. Verify you can see the new `Marketing` tab.
4. Click that tab.
5. Verify you can see the subtabs.
6. Click around the subtabs. Verify that the URL changes.
7. Click the second subtab (Business Tools) then refresh the page. Verify that the URL works for a fresh page load.

> [!WARNING]  
> The mobile view design is still not implemented in this PR; will follow-up later. Including more polishes to the Desktop version.

## Pre-merge Checklist
